src/test/java/com/google/javascript/gents/singleTests/fn_types.js

```diff
@@ -47,3 +47,9 @@ var restParamsTyped = function(n, br) {};
  * @type {function(number, boolean=, ...): number}
  */
 const complex = function(n, o, r) { return n; };
+
+/** @type {?function():string} */
+let f1 = null;
+
+/** @type {function()|string} */
+const f2 = 'string';
```

Comment:
Add something trickier like

`function():string|function():string` -> `() => string | () => string`

Reply:
That's just `(function():string)|(function(): string)`, which is `function():string`.
Did you mean `function():(string|function():string)`?

I added test cases for both anyway.